src/backend/distributed/operations/shard_split.c

```diff
@@ -427,13 +427,26 @@ SplitShard(SplitMode splitMode,
 							errmsg("Invalid Node Id '%u'.", nodeIdValue)));
 		}
 
+		if (!NodeHasAllReferenceTableReplicas(workerNode))
+		{
```

Comment:
Minor: 
There could be repeated destination workers and we might end-up checking for the same node again and again for referecnce table existence. 
It would be good to use `ListToHashSet` and call the `NodeHasAllReferenceTableReplicas` on each worker node exactly once.